packages/tizen_rpc_port/example/example_proxy/lib/message_proxy.dart

```diff
@@ -0,0 +1,262 @@
+
+    await onReceived(sender, msg);
+  }
+}
+
+/// Abstract class for creating [Message] class for RPC.
+abstract class Message extends ProxyBase {
+  /// Constructor for this class.
+  Message(String appid) : super(appid, 'Message');
+
+  bool _online = false;
+  final List<CallbackBase> _delegateList = <CallbackBase>[];
+
+  /// This abstract method will be called when the connet() is succeed.
+  Future<void> onConnected();
+
+  /// This abstract method will be called when the connection with the stub is disconnected.
+  Future<void> onDisconnected();
+
+  /// This abstract method will be called when connect() is failed.
+  Future<void> onRejected(String errorMessage);
+
+  @override
+  @nonVirtual
+  Future<void> onConnectedEvent() async {
+    _online = true;
+    await onConnected();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onDisconnectedEvent() async {
+    _online = false;
+    await onDisconnected();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onRejectedEvent(String errorMessage) async {
+    await onRejected(errorMessage);
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onReceivedEvent(Parcel parcel) async {
+    final int cmd = parcel.readInt32();
+    if (cmd != _MethodId.callback.id) {
+      return;
+    }
+
+    await _processReceivedEvent(parcel);
+  }
+
+  Future<void> _processReceivedEvent(Parcel parcel) async {
+    final int id = parcel.readInt32();
+    final int seqId = parcel.readInt32();
+    final bool once = parcel.readBool();
+
+    for (final CallbackBase delegate in _delegateList) {
+      if (delegate._id == id && delegate._seqId == seqId) {
+        await delegate._onReceivedEvent(parcel);
+        if (delegate._once && once) {
+          _delegateList.remove(delegate);
+        }
+        break;
+      }
+    }
+  }
+
+  Future<Parcel> _consumeCommand(Port port) async {
+    while (true) {
+      try {
+        final Parcel parcel = Parcel.fromPort(port);
+        final int cmd = parcel.readInt32();
+        if (cmd == _MethodId.result.id) {
+          return parcel;
+        }
+      } catch (e) {
+        Log.error(_logTag, e.toString());
+        return Parcel();
+      }
+    }
+  }
+
+  /// Connects with the stub application.
+  @override
+  Future<void> connect() async {
+    Log.info(_logTag, 'connect()');
+    await super.connect();
+  }
+
+  /// Dispose registered delegate interface.
+  void disposeCallback(String tag) {
+    Log.info(_logTag, 'disposeCallback($tag)');
+    _delegateList.removeWhere((CallbackBase element) => element.tag == tag);
+  }
+
+  /// This method is used to send 'Register' request to the stub app.
+  Future<int> register(String name, NotifyCB cb) async {
+    Log.info(_logTag, 'Register');
+
+    if (!_online) {
+      throw Exception('NotConnectedSocketException');
+    }
+
+    final Port port = getPort(PortType.main);
+
+    final Parcel parcel = Parcel();
+    final ParcelHeader header = parcel.header;
+    header.tag = _tidlVersion;
```

Comment:
Does a ParcelHeader tag always represent a TIDL version? (As far as I see the header can have only a single tag.)

Reply:
Files generated by TIDL always store versions.
It exists to match the compatibility with the old version.

Reply:
I know what it is for but I wonder why it's named "tag" but not "version". (This is just a question. I'm not requesting a change.)

Reply:
rpc port API is almost used by TIDL, but in some cases, only the the API may be used. So, to some extent, it can be said to be for versatility.